src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -35,296 +35,650 @@
+        if (accountsToRename.size() > 0) {
+            for (Account acc : accountsToRename) {
+                acc.Name = UTIL_ACCT_Naming.updateName(acc.Contacts);
+            }
+        }
+
+        //For now, we use direct DML statement to avoid duplicate id issue
+        update accountsToRename;
+    }
+
+    /*******************************************************************************************************
+    * @description Retrieves additional Contact and Account information.
+    * @param accIdsToRename a set of AccountIds that needs Name update.
+    * @return List<Account>.
+    */
+    private static List<Account> queryContacts(Set<Id> accIdsToRename) {
+        List<Account> returnedContacts = new List<Account>();
+        String sortByLastName = 'LastName';
+        String sortByFirstName = 'FirstName';
 
         //Build dynamic query string
-        string strSoql = 'SELECT id, RecordTypeId, ';
-        strSoql += '(SELECT id, AccountId,' +
-                'Account.RecordTypeID, ' +
-                'Account.Primary_Contact__c, Account.Name,' +
-                'firstname, lastname, OwnerId, Salutation,' +
-                'MailingStreet, MailingCity, MailingState, MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, ' +
-                'OtherStreet, OtherCity, OtherState, OtherPostalCode, OtherCountry, OtherLatitude, OtherLongitude, ';
+        String dynamicSoql = 'SELECT id, RecordTypeId, ';
+        dynamicSoql += '(SELECT Id, AccountId,' +
+            'Account.RecordTypeID, ' +
+            'Account.Primary_Contact__c, Account.Name,' +
+            'firstname, lastname, OwnerId, Salutation,' +
+            'MailingStreet, MailingCity, MailingState, ' +
+            'MailingPostalCode, MailingCountry, MailingLatitude, ' +
+            'MailingLongitude, OtherStreet, OtherCity, OtherState, ' +
+            'OtherPostalCode, OtherCountry, OtherLatitude, OtherLongitude, ';
+
+        if (ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) {
+            dynamicSoql += 'MailingCountryCode, MailingStateCode, OtherCountryCode, OtherStateCode, ';
+        }
 
-        if (ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled)
-            strSoql += 'MailingCountryCode, MailingStateCode, OtherCountryCode, OtherStateCode, ';
+        if (defaultRecTypeId == hhAccountRecordTypeId) {
+            dynamicSoql += 'Phone, Fax FROM Contacts WHERE Exclude_from_Household_Name__c != true)';
+        } else {
+            dynamicSoql += 'FROM Contacts)';
+        }
 
-        strSoql += 'Phone, Fax FROM Contacts WHERE Exclude_from_Household_Name__c != true)';
-        strSoql += 'FROM Account WHERE Id IN :accIdsToRename';
+        dynamicSoql += 'FROM Account WHERE Id IN :accIdsToRename';
 
         //Re-query to get correct Account values (and all the other fields we will look at)
-        listAccsToRename = database.query(strSoql);
+        returnedContacts = database.query(dynamicSoql);
```

Comment:
consider using String.escapeSingleQuotes on the dynamicSoql query. The platform already does this for fields but it shows that we are indeed leveraging best practice.

Reply:
@lifewithryan  yep! Updated it to add that.